pageserver/src/disk_usage_eviction_task.rs

```diff
@@ -390,24 +390,29 @@ pub async fn disk_usage_eviction_task_iteration_impl<U: Usage>(
                     assert_eq!(results.len(), batch.len());
                     for (result, layer) in results.into_iter().zip(batch.iter()) {
                         match result {
-                            Some(Ok(true)) => {
+                            Some(Ok(())) => {
                                 usage_assumed.add_available_bytes(layer.file_size());
                             }
-                            Some(Ok(false)) => {
-                                // this is:
-                                // - Replacement::{NotFound, Unexpected}
-                                // - it cannot be is_remote_layer, filtered already
+                            Some(Err(EvictionError::CannotEvictRemoteLayer)) => {
+                                unreachable!("checked already")
```

Comment:
this is because we do the collection with `Timeline::get_local_layers_for_disk_usage_eviction`. I think the "local layers" in the name. But I should move this rationale to the message.

Reply:
[72dd218](https://github.com/neondatabase/neon/pull/4737/commits/72dd2188b40e8173f023d57c4012b9c8729b6403)